test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildServerIntegrationTest.cs

```diff
@@ -143,28 +144,55 @@ public async Task Build_ServerConnectionMutexCreationFails_FallsBackToInProcessR
             // it reaches server creation part.
         }
 
-        // Skipping on linux/mac because of https://github.com/aspnet/Razor/issues/2507.
+        // Skipping on MacOS because of https://github.com/dotnet/corefx/issues/33141.
         [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
         [InitializeTestProject("SimpleMvc")]
         public async Task ManualServerShutdown_NoPipeName_ShutsDownServer()
         {
-            var toolAssembly = typeof(Application).Assembly.Location;
-            var result = await DotnetMSBuild(
-                "Build",
-                $"/p:_RazorForceBuildServer=true /p:_RazorToolAssembly={toolAssembly}",
-                suppressBuildServer: true); // We don't want to specify a pipe name
+            // Publish rzc tool to a temporary path
+            var solutionRoot = TestPathUtilities.GetSolutionRootDirectory("Razor");
+            var toolAssemblyDirectory = Path.Combine(solutionRoot, "src", "Microsoft.AspNetCore.Razor.Tools");
+            var toolAssemblyPath = Path.Combine(toolAssemblyDirectory, "Microsoft.AspNetCore.Razor.Tools.csproj");
+            var projectDirectory = new TestProjectDirectory(solutionRoot, toolAssemblyDirectory, toolAssemblyPath);
+            var publishDir = Path.Combine(Path.GetTempPath(), "Razor", Path.GetRandomFileName(), "RzcPublish");
+            var publishResult = await MSBuildProcessManager.RunProcessAsync(projectDirectory, $"/t:Publish /p:PublishDir=\"{publishDir}\"");
```

Comment:
You should verify that this succeeds and make sure the directory gets cleaned up.

Reply:
Basically add a verification here and move inside the `try`

Reply:
I'll add the assert. I'm already cleaning it up in `finally`